### PR TITLE
First iteration for mock server(s)

### PR DESCRIPTION
**Pull request by gunzip**
- splitted notifications specs in 'public' and 'private'
- added swagger-express mock endpoints (server.js)
- added npm scripts in package.json to edit and lint swagger specs

- added editorconfig
- added eslint rules

- added simple sendMessage / getMessage NodeJS client
- added lokijs in-memory persistence for fake data (fixtures)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,19 @@
-# Cittadinanza Digitale
+# Digital Citizenship Initiative
 
 This is work in progress.
```
Comment (cloudify) [inline README.md]: Perhaps here we can link at the one pager to learn more

File: docs/server.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+If you want to experience API usage in a test environment,
+in addition to the API documentation, you can download and use a NodeJS (express)
+HTTP server which implements the APIs based on the OpenAPI specifications.
+
+## Downloading the test environment
+
+Start cloning the GitHub repository:
+
+`git clone http://github.com/teamdigitale/cittadinanza-digitale`
```
Comment (cloudify) [inline docs/server.md]: use  multi-line block here with triple `

File: docs/server.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+`cd cittadinanza-digitale`
+`npm install`
+
+## Starting the mock server
+
+Type
+
+`npm run serve`
```
Comment (cloudify) [inline docs/server.md]: also here a multi line block, it's good practice to use single backtick code blocks only inline

File: docs/server.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+though a tool like [Postman](https://www.getpostman.com/) or with the help of
+the embedded interface (swagger-editor):
+
+`npm run edit`
+
+## Running the documentation website locally
+
+To run the documentation hub locally:
+
+`mkdocs serve` to run documentation site locally
```
Comment (cloudify) [inline docs/server.md]: bullet points would be good here

File: mock/client/config.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+'use strict'
+
+const config = {
+  NOTIFICATION_API_URL: 'http://localhost:10010/api/v1'
```
Comment (cloudify) [inline mock/client/config.js]: needs comment about what this URL is and why would you have to change it

File: mock/client/getMessage.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+'use strict'
```
Comment (cloudify) [inline mock/client/getMessage.js]: in general it's good to have block comments at the beginning of each source file that explain the purpose of that file

File: mock/client/sendMessage.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+'use strict'
+
+const config = require('./config')
```
Comment (cloudify) [inline mock/client/sendMessage.js]: Think about this code as documentation, we need a lot more comments here that explain what's going on.

**Comment by gunzip** [comment]
updated. i would start merging this into master then we can switch from 'notifications API' to 'messaging API' with another pr.

**Pull request closed without merging**